certs_net.py

```diff
@@ -8,61 +8,115 @@
 import ssl, socket
 import datetime
 import time
+import sys
 
 urllib3.disable_warnings()
-logger = logging.getLogger(__name__)
-
-
 
 class CertsNet(object):
     namespace = None
     use_kubeconfig = True
     trawler = None
+    logger = None
 
-    def __init__(self, config, trawler):
+    def __init__(self, config, trawler=None, logger=None):
         # Takes in config object and trawler instance it's behind
         # Use kubeconfig or in-cluster config for k8s comms
         if trawler:
             self.trawler = trawler
-            self.use_kubeconfig = trawler.use_kubeconfig
+            self.use_kubeconfig = self.trawler.use_kubeconfig
+        if logger:
+            self.logger = logger
+        else:
+            alog.configure(default_level="info")
+            self.logger = alog.use_channel("certs_net")
+        self.logger.info("Init CertsNet")
         # Namespace to review
         self.namespace = config.get('namespace', None)
 
+
     def getExpiry(self, cert_data):
-      cert = base64.b64decode(cert_data)
-      x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, cert)
-      expiry = datetime.datetime.strptime(x509.get_notAfter().decode('utf-8'), "%Y%m%d%H%M%S%z").timestamp()
-      return int(expiry - time.time())
+        try:
+            cert = base64.b64decode(cert_data)
+            x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, cert)
+            expiry = datetime.datetime.strptime(x509.get_notAfter().decode('utf-8'), "%Y%m%d%H%M%S%z").timestamp()
+            return int(expiry - time.time())
+        except:
+            self.logger.exception("getExpiry failed")
```

Comment:
We should probably capture the exception detail and pass to the logger here

Reply:
there's a couple of other logger.exceptions that could do with the same handling

Reply:
> We should probably capture the exception detail and pass to the logger here

Doesn't logging.exception() do that anyway? What more needs to be output?
>exception(msg, *args, **kwargs)
Logs a message with level ERROR on this logger. The arguments are interpreted as for [debug()](https://docs.python.org/3/library/logging.html#logging.debug). Exception info is added to the logging message. This method should only be called from an exception handler.

Reply:
ah I'd missed that - all good then